src/pam/meson.build

```diff
@@ -1,9 +1,25 @@
 project('pam_howdy', 'cpp', version: '0.1.0', default_options: ['cpp_std=c++14'])
 
-inih = subproject('inih')
-inih_cpp = inih.get_variable('INIReader_dep')
+inih_cpp = dependency('INIReader', required: false)
 
-libpam = meson.get_compiler('c').find_library('pam')
+if not inih_cpp.found()
+	inih_cpp = subproject('inih').get_variable('INIReader_dep')
```

Comment:
inih_cpp will be found from the wrap file, just use `required: true` and drop the manual subproject call.

That is what
```ini

[provide]
inih = inih_dep
inireader = inireader_dep
```
is there for.

Reply:
Thanks for the suggestion. That's because I don't want to init the subproject if the inireader found on system wide. In some areas, it could take a long time.

Reply:
It's not entirely clear to me that you realize what I'm saying?

Meson *already* uses the system wide copy, if it can find it.

Meson *already* inits the subproject, ***if it cannot find the system wide copy***.

This is a builtin feature of meson, and one of the core strengths of the Wrap Dependency subsystem as compared to other build systems. You don't need to do extra work. Meson just does the right thing out of the box. That means not doing pointless work like initing the subproject, when meson detects that it doesn't need to (because there is a system copy).

Reply:
> It's not entirely clear to me that you realize what I'm saying?

Maybe.

> Meson already uses the system wide copy, if it can find it.

Yes, it is.

> Meson already inits the subproject, if it cannot find the system wide copy.

That's why I want the subproject just to fallback.

This change is base on <https://mesonbuild.com/Subprojects.html#toggling-between-system-libraries-and-embedded-sources>.

*BTW, I'm actually not a specialist of Meson.*